pkg/cosign/tuf/client.go

```diff
@@ -114,21 +81,22 @@ func New(ctx context.Context, remote client.RemoteStore, cacheRoot string) (*TUF
 	}
 
 	// We need to update our tufdb.
-	// Warning: If a local cache already exists, you may get a local/remote mismatch
-	// since the default remote may not match the remote repository configured during
-	// a cosign initialize.
 	trustedRoot, err := getRoot(trustedMeta)
 	if err != nil {
+		t.local.Close()
```

Comment:
nit: You can call `t.Close()`, which currently just wraps `t.local.Close()`, but maybe we'll need to extend it at some point